dashboard/test/ui/step_definitions/publicKeyCryptography_steps.rb

```diff
@@ -33,9 +33,8 @@
   @browser.find_element(:css, '.go-button').click
 end
 
-When /^I (?:hide|show) Eve's panel$/ do
-  @browser.find_element(:xpath, "//div[text()='Eve']").click
-  sleep 0.5
+When /^I open view "(Alice|Eve|Bob|All)"$/ do |view|
+  @browser.find_element(:xpath, "//button[text()[contains(.,'#{view}')]]").click
```

Comment:
The xpath selector we had trouble with was an exact-text match for a control that also contained an icon.  I specifically worked out how to look up the thing that 'contains' the text here, not an exact match for the text content.  I don't think there's a more general problem with xpath, except that as a team we don't know it particularly well... but it's the default for selenium in general, and since these aren't jquery buttons I'd rather do it this way.